include/oneapi/dpl/internal/function.h

```diff
@@ -90,6 +107,190 @@ get_access(const Policy& policy, T* ptr)
     assert(sycl::get_pointer_type(ptr, policy.queue().get_context()) == sycl::usm::alloc::shared);
     return ptr;
 }
+
+template <typename Policy, typename T>
+auto
+get_buffer_for(const Policy& policy, T* ptr)
+    -> decltype(sycl::buffer(ptr, 1))
+{
+    return sycl::buffer(ptr, 1);
+}
+
+template <typename ValueType, typename Policy, typename Iterator>
+void
+get_data_at(const Policy& p, const Iterator& i, size_t index, ValueType& val)
+{
+    sycl::queue queue = p.queue();
+
+    sycl::buffer<ValueType, 1> host_buffer(&val, 1);
+    auto src_buffer = get_buffer_for(p, i);
+
+    queue.submit(
+        [&](sycl::handler& cgh)
+        {
+            auto src_buffer_acc = src_buffer.template get_access<sycl::access::mode::read>(cgh);
+            auto host_buffer_acc = host_buffer.template get_access<sycl::access::mode::write>(cgh);
+
+            cgh.parallel_for(sycl::range<1>(1),
```

Comment:
I think that usage of SYCL "singe_task" is more preferable than "trivial" parallel_for with n=1.

Reply:
Fixed